packages/expo-cli/src/commands/eas-build/utils/expoUpdates.ts

```diff
@@ -0,0 +1,370 @@
+  const pbxprojPaths = await new Promise<string[]>((resolve, reject) =>
+    glob('ios/*/project.pbxproj', { absolute: true, cwd: projectDir }, (err, res) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    })
+  );
+
+  const pbxprojPath = pbxprojPaths.length > 0 ? pbxprojPaths[0] : undefined;
+
+  if (!pbxprojPath) {
+    throw new Error("Couldn't find Xcode project");
+  }
+
+  return pbxprojPath;
+}
+
+async function getXcodeProjectAsync(pbxprojPath: string) {
+  const project = xcode.project(pbxprojPath);
+
+  await new Promise((resolve, reject) =>
+    project.parse(err => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    })
+  );
+
+  return project;
+}
+
+function getExpoPlistPath(projectDir: string, pbxprojPath: string) {
+  const xcodeprojPath = path.resolve(pbxprojPath, '..');
+  const expoPlistPath = path.resolve(
+    projectDir,
+    'ios',
+    path.basename(xcodeprojPath).replace(/\.xcodeproj$/, ''),
+    'Supporting',
+    'Expo.plist'
+  );
+
+  return expoPlistPath;
+}
+
+async function getBundleReactNativePhaseAsync(project: xcode.XcodeProject) {
+  const scriptBuildPhase = project.hash.project.objects.PBXShellScriptBuildPhase;
+  const bundleReactNative = Object.values(scriptBuildPhase).find(
+    buildPhase => buildPhase.name === '"Bundle React Native code and images"'
+  );
+
+  if (!bundleReactNative) {
+    throw new Error(`Couldn't find a build phase script for "Bundle React Native code and images"`);
+  }
+
+  return bundleReactNative;
+}
+
+async function configureUpdatesAndroidAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  username: string | null
+) {
+  const buildGradlePath = getAndroidBuildGradlePath(projectDir);
+  const buildGradleContent = await getAndroidBuildGradleContentAsync(buildGradlePath);
+
+  if (!hasBuildScriptApply(buildGradleContent)) {
+    await fs.writeFile(
+      buildGradlePath,
+      `${buildGradleContent}\n// Integration with Expo updates\n${androidBuildScript}\n`
+    );
+  }
+
+  const androidManifestPath = getAndroidManifestPath(projectDir);
+  const androidManifestJSON = await AndroidConfig.Manifest.readAndroidManifestAsync(
+    androidManifestPath
+  );
+
+  if (!isMetadataSetAndroid(androidManifestJSON, exp, username)) {
+    const result = await AndroidConfig.Updates.setUpdatesConfig(exp, androidManifestJSON, username);
+
+    await AndroidConfig.Manifest.writeAndroidManifestAsync(androidManifestPath, result);
+  }
+}
+
+async function isUpdatesConfiguredAndroidAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  username: string | null
+) {
+  const buildGradlePath = getAndroidBuildGradlePath(projectDir);
+  const buildGradleContent = await getAndroidBuildGradleContentAsync(buildGradlePath);
+
+  if (!hasBuildScriptApply(buildGradleContent)) {
+    return false;
+  }
+
+  const androidManifestPath = getAndroidManifestPath(projectDir);
```

Comment:
```suggestion
  const androidManifestPath = await AndroidConfig.Manifest.getProjectAndroidManifestPathAsync(
    projectDir
  );
  if (!androidManifestPath) {
    throw new Error(`Could not find AndroidManifest.xml in project directory: "${projectDir}"`);
  }
```
same as `expo-cli/src/commands/init`